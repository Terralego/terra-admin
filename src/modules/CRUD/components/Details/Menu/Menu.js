import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import classnames from 'classnames';
import {
  Alignment,
  Classes,
  Navbar,
  NavbarGroup,
  Icon,
} from '@blueprintjs/core';

import { generateURI } from '../../../config';
import DeleteFeature from '../DeleteFeature';

const nav = [
  { section: 'default', text: 'CRUD.details.menu.default', icon: 'list-detail-view' },
  { section: 'geometries', text: 'CRUD.details.menu.geometries', icon: 'polygon-filter' },
  { section: 'attachments', text: 'CRUD.details.menu.attachments', icon: 'paperclip' },
];

const Menu = ({
  section,
  match: { params: { id, layer } },
  t,
}) => (
  <Navbar className="details__menu">
    <NavbarGroup>
      {nav.map(({ icon, text, ...props }) => (
        <NavLink
          key={props.section}
          to={generateURI('layer', { layer, id, section: props.section })}
        >
          <span className={classnames({
            [Classes.BUTTON]: true,
            [Classes.MINIMAL]: true,
            [Classes.ACTIVE]: props.section === section,
          })}
          >
            {icon && <Icon icon={icon} />}
            <span className="bp3-button-text"> {t(text)}</span>
          </span>
        </NavLink>
      ))}
    </NavbarGroup>
    <NavbarGroup align={Alignment.RIGHT}>
      <DeleteFeature />
    </NavbarGroup>
  </Navbar>
);

Menu.propTypes = {
  section: PropTypes.oneOf(['default', 'geometries', 'attachments']),
  match: PropTypes.shape({
    params: PropTypes.shape({
      layer: PropTypes.string,
      id: PropTypes.string,
      action: PropTypes.string,
    }),
  }),
  t: PropTypes.func,
};

Menu.defaultProps = {
  section: 'default',
  match: {
    params: {
      layer: undefined,
      id: undefined,
    },
  },
  t: text => text,
};

export default Menu;
