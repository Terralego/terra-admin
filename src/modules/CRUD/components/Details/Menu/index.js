import { withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';

import Menu from './Menu';
import compose from '../../../../../utils/compose';

export default compose(
  withRouter,
  withTranslation(),
)(Menu);
