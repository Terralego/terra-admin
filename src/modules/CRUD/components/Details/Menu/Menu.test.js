import React from 'react';
import renderer from 'react-test-renderer';

import Menu from './Menu';

jest.mock('react-router-dom', () => ({
  NavLink: ({ children, ...props }) => <a {...props}>{children}</a>,
}));

jest.mock('@blueprintjs/core', () => ({
  NavbarGroup: ({ children }) => <div>{children}</div>,
  Navbar: ({ children }) => <div>{children}</div>,
  Icon: ({ icon }) => <div>{icon}</div>,
  Alignment: {
    RIGHT: 'right',
  },
  Classes: {
    BUTTON: 'btn',
    MINIMAL: 'minimal',
    ACTIVE: 'active',
    DISABLED: 'disabled',
  },
}));

jest.mock('../../../config', () => ({
  generateURI: jest.fn(),
}));

jest.mock('../DeleteFeature', () => ({ children, ...props }) => <div {...props}>{children}</div>);

const props = {
  section: 'default',
  match: { params: { layer: 'layerFoo', id: 'featureFoo' } },
};

it('should render correctly', () => {
  const tree = renderer.create((
    <Menu
      {...props}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
