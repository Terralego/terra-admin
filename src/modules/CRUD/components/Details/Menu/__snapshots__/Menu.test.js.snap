// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <a>
      <span
        className="btn minimal active"
      >
        <div>
          list-detail-view
        </div>
        <span
          className="bp3-button-text"
        >
           
          CRUD.details.menu.default
        </span>
      </span>
    </a>
    <a>
      <span
        className="btn minimal"
      >
        <div>
          polygon-filter
        </div>
        <span
          className="bp3-button-text"
        >
           
          CRUD.details.menu.geometries
        </span>
      </span>
    </a>
    <a>
      <span
        className="btn minimal"
      >
        <div>
          paperclip
        </div>
        <span
          className="bp3-button-text"
        >
           
          CRUD.details.menu.attachments
        </span>
      </span>
    </a>
  </div>
  <div>
    <div />
  </div>
</div>
`;
